__TESTS__/backwardsComaptibility/createV1URL.test.ts

```diff
@@ -81,6 +81,23 @@ describe('Create v1 urls', () => {
     expect(url).toEqual("https://res.cloudinary.com/test123/image/upload/if_w_lt_200,c_fill,h_120,w_80/if_w_gt_400,c_fit,h_150,w_150/e_sepia/sample");
   });
 
+  it("should not modify options objects when chaining transformations ", function () {
+    const optionObj = {
+      cloud_name: 'test123',
+      transformation: [{
+        overlay: 'somepid',
+        crop: "fill",
+        height: 120,
+        width: 80
+      }]
+    };
+    const urlFirstRun = createCloudinaryV1URL("sample", optionObj);
```

Comment:
This behaviour should match the v1 behaviour, where the options are mutated as they're being consumed by the API.

We'll confirm if there's a regression from v1 and we'll deploy a fix if needed.

Reply:
I see, that makes sense since I used the same options objects in V1 and V2 one after the other this highlighted that point. Indeed this is not the normal use case to "parse" the same options object twice.

How do you feel about adding V1 as a dev dependency and using it in tests to get a better compatibility test coverage?

This way one could build a simple infrastructure that does (pseudo code):
```
(pubid, options, config) => expect(
    v2url(pubid, {...config, ...options})
  ).toEqual(
    v1url(config)(pubid, options)
)
```
And adding the test cases would be a breeze - covering 100's of cases with very little effort.